filesystem/iso9660.ksy

```diff
@@ -1,210 +1,344 @@
+      - id: system_identifier
+        doc-ref: ecma-119 8.5.4
+        type: str
+        size: 0x20
+        encoding: ASCII
+      - id: volume_identifier
+        doc-ref: ecma-119 8.5.5
+        type: str
+        size: 0x20
+        encoding: ASCII
+      - id: unused01
+        doc-ref: ecma-119 8.5
+        size: 0x8
+      - id: volume_space_size
+        doc-ref: ecma-119 8.5
         type: u4bi
-      - id: datetime
-        type: datetime
-      - id: file_flags
-        type: u1
-      - id: file_unit_size
-        type: u1
-      - id: interleave_gap_size
-        type: u1
-      - id: vol_seq_num
+      - id: escape_sequences
+        doc-ref: ecma-119 8.5.6
+        size: 0x20
+      - id: volume_set_size
+        doc-ref: ecma-119 8.5
         type: u2bi
-      - id: len_file_name
-        type: u1
-      - id: file_name
-        type: str
-        encoding: UTF-8
-        size: len_file_name
-      - id: padding
-        type: u1
-        if: len_file_name % 2 == 0
-      - id: rest
-        size-eos: true
-    instances:
-      extent_as_dir:
-        io: _root._io
-        pos: lba_extent.le * _root.sector_size
-        size: size_extent.le
-        type: dir_entries
-        if: file_flags & 2 != 0
-      extent_as_file:
-        io: _root._io
-        pos: lba_extent.le * _root.sector_size
-        size: size_extent.le
-        if: file_flags & 2 == 0
-  ## AKA "Path Table Entry"
-  path_table_le:
-    doc-ref: 'http://wiki.osdev.org/ISO_9660#The_Path_Table'
-    seq:
-      - id: entries
-        type: path_table_entry_le
-        repeat: eos
-  path_table_entry_le:
-    seq:
-      - id: len_dir_name
-        type: u1
-      - id: len_ext_attr_rec
-        type: u1
-      - id: lba_extent
+      - id: volume_sequence_number
+        doc-ref: ecma-119 8.5
+        type: u2bi
+      - id: logical_block_size
+        doc-ref: ecma-119 8.5
+        type: u2bi
+      - id: path_table_size
+        doc-ref: ecma-119 8.5.7
+        type: u4bi
+      - id: occurrence_of_type_l_path_table
+        doc-ref: ecma-119 8.5.8
         type: u4le
-      - id: parent_dir_idx
-        type: u2le
-      - id: dir_name
+      - id: optional_occurrence_of_type_l_path_table
+        doc-ref: ecma-119 8.5.9
+        type: u4le
+      - id: occurrence_of_type_m_path_table
+        doc-ref: ecma-119 8.5.10
+        type: u4le
+      - id: optional_occurrence_of_type_m_path_table
+        doc-ref: ecma-119 8.5.11
+        type: u4le
+      - id: directory_record_for_root_directory
+        doc-ref: ecma-119 8.5.12
+        type: directory_record
+        size: 0x22
+      - id: volume_set_identifier
+        doc-ref: ecma-119 8.5.13
         type: str
-        encoding: UTF-8
-        size: len_dir_name
-      - id: padding
-        type: u1
-        if: len_dir_name % 2 == 1
-  datetime:
+        size: 0x80
+        encoding: ASCII
+      - id: publisher_identifier
+        doc-ref: ecma-119 8.5.14
+        type: str
+        size: 0x80
```

Comment:
Consider moving these stings of predefined lengths into own types

Reply:
Done